halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/model/v1/node/Node.java

```diff
@@ -206,4 +213,130 @@ public String getName() {
   public String debugName() {
     return "[" + getClass().getSimpleName() + ":" + getNodeName() + "]";
   }
+
+
+  private Map<String, Object> serializedNonNodeFields() {
+    List<Field> fields = Arrays.stream(this.getClass().getDeclaredFields()).filter(f -> {
+      if (Node.class.isAssignableFrom(f.getType())) {
+        return false;
+      }
+
+      if (List.class.isAssignableFrom(f.getType())) {
+        return false;
+      }
+
+      if (Map.class.isAssignableFrom(f.getType())) {
+        return false;
+      }
+
+      if (f.getAnnotation(JsonIgnore.class) != null) {
+        return false;
+      }
+
+      return true;
+      }).collect(Collectors.toList());
+
+    Map<String, Object> res = new HashMap<>();
+    for (Field field : fields) {
+      field.setAccessible(true);
+      try {
+        res.put(field.getName(), field.get(this));
+      } catch (IllegalAccessException e) {
+        throw new RuntimeException("Failed to read field " + field.getName() + " in node " + getNodeName());
+      }
+      field.setAccessible(false);
+    }
+
+    return res;
+  }
+
+  private Map<String, Node> serializedNodeFields() {
+    Map<String, Node> res = new HashMap<>();
+
+    NodeIterator iterator = this.getChildren();
+    Node next = iterator.getNext();
+
+    while (next != null) {
+      res.put(next.getNodeName(), next);
+      next = iterator.getNext();
+    }
+
+    return res;
+  }
+
+  public NodeDiff diff(Node other) {
+    String tt = this.getClass().getSimpleName();
+    String to = other.getClass().getSimpleName();
+    if (!tt.equals(to)) {
+      throw new RuntimeException("Invalid comparision between unequal node types (" + tt + " != " + to + ")");
+    }
+
+    String nnt = this.getNodeName();
+    String nno = other.getNodeName();
+    if (!nnt.equals(nno)) {
+      throw new RuntimeException("Invalid comparision between different nodes (" + nnt + " != " + nno + ")");
+    }
+
+    NodeDiff result = new NodeDiff().setChangeType(EDITED).setNode(this);
+
+    Map<String, Object> fts = this.serializedNonNodeFields();
+    Map<String, Object> fos = other.serializedNonNodeFields();
+
+    for (Map.Entry<String, Object> entry : fts.entrySet()) {
+      String fnt = entry.getKey();
+      Object ot = entry.getValue();
```

Comment:
They're all acronyms, the last char is `t` or `o` for `this` or `other`, the prefix is the kind of value. The variable names grow too long too quickly for such short-lived values IMO.